### PR TITLE
Feature/699 sri implement circuit from workstep 1 from the sri use case docs

### PR DESCRIPTION
**Pull request by biscuitdey**
# Description

This PR prepares the inputs for circuit execution as well as allows dynamic input of circuits in different worksteps.

## Related Issue

#699 #701 

## Motivation and Context

Prepares for circuit execution

## How Has This Been Tested

Manual test by running npm run test and npm run build. More extensive testing, both manual and automated, will follow.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM

File: examples/bri-3/zeroKnowledgeKeys/scripts/circuit.sh
Comment (ognjenkurtic) [inline examples/bri-3/zeroKnowledgeKeys/scripts/circuit.sh:1]: All of these scripts that perform compilation of the circuit, zkeys and accompanying infrastructure are not something that should be part of the bri-3 source code. In my eyes, these artifacts should be supplied compiled by the workgroup participants and in mil5 we need to implement infrastructure that allow for that. For now, i would place them into a separate folder, i.e. examples/bri-3/sample-use-case/artifacts...

File: examples/bri-3/zeroKnowledgeKeys/scripts/ptau.sh [hunk 1]
```diff
@@ -8,38 +8,38 @@ set -e
 # make ptau folder
 [ -d zeroKnowledgeKeys/ptau ] || mkdir zeroKnowledgeKeys/ptau
 
-# Starts Powers Of Tau ceremony, creating the file pot14_0000.ptau
-# 14 is the power of two of the maximum number of constraints that the ceremony can accept: in this case, the number of constraints is 2 ^ 14 = 16,384.
-# 14 is required as ecdsaSignatureVerification takes 8,076 constraints
-snarkjs powersoftau new bn128 14 zeroKnowledgeKeys/ptau/pot14_0000.ptau -v
+# Starts Powers Of Tau ceremony, creating the file pot20_0000.ptau
```
Comment (ognjenkurtic) [inline examples/bri-3/zeroKnowledgeKeys/scripts/ptau.sh:11]: could we just use a readily available ptau file from the web for the sample use-case or we need to go through the ceremony?

**Review by ognjenkurtic — APPROVED** [review]

File: examples/bri-3/zeroKnowledgeKeys/scripts/circuit.sh
Comment (Therecanbeonlyone1969) [reply examples/bri-3/zeroKnowledgeKeys/scripts/circuit.sh:1]: @ognjenkurtic there is a workflow that snarkjs or gnark for that matter must go through. All of that could be supplied during set-up but it is much earlier if the workflow is automated as it comes with builtin scripts as used by @biscuitdey . The only thing that needs to be supplied is the circuit file (in this case the circom file).

**Review by Kasshern — APPROVED** [review]

File: examples/bri-3/zeroKnowledgeKeys/scripts/ptau.sh (hunk 1 above)
Comment (Therecanbeonlyone1969) [reply examples/bri-3/zeroKnowledgeKeys/scripts/ptau.sh:11]: if groth16 is used then you need to run a trusted set-up for every invocation of a circuit with public and private inputs. That is not necessary with PLONK. There we can use an available common reference string (CRS) from the public power of tau ceremony that has already been done. So you see it depends what commitment schemes are used.

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts [hunk 2]
```diff
@@ -2,37 +2,37 @@ import { Injectable } from '@nestjs/common';
...
+    );
 
     this.witness.proof = proof;
 
     this.witness.publicInputs = publicInputs;
 
-    // TODO: Above TODO
-    // this.witness.verificationKey = verificationKey;
+    this.witness.verificationKey = await import(
+      `../../../../../../zeroKnowledgeKeys/circuit/${circuitName}_verification_key.json`
```
[the remark this answers is not in the document]
Comment (biscuitdey) [reply examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts:31]: @ognjenkurtic Sure you can add the path in the config file in your PR.

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts [hunk 3]
```diff
@@ -50,12 +50,13 @@ export class SnarkjsCircuitService implements ICircuitService {
...
   private async executeCircuit(
     inputs: object,
+    circuitName: string,
   ): Promise<{ proof: Proof; publicInputs: string[] }> {
     const { proof, publicSignals: publicInputs } =
       await snarkjs.groth16.fullProve(
         inputs,
-        process.env.SNARKJS_CIRCUIT_WASM,
-        process.env.SNARKJS_PROVING_KEY,
+        `zeroKnowledgeKeys/circuit/${circuitName}_js/${circuitName}.wasm`,
```
[the remark this answers is not in the document]
Comment (biscuitdey) [reply examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts:58]: @ognjenkurtic Sure you can add the path in the config file in your PR.

File: examples/bri-3/zeroKnowledgeKeys/scripts/circuit.sh
Comment (biscuitdey) [reply examples/bri-3/zeroKnowledgeKeys/scripts/circuit.sh:1]: @ognjenkurtic The circuit.sh file is not a sample file. Using the `npm run snarkjs:circuit {circuitName}`, the user can specify their own `circuitName` as a parameter, and circuit.sh will dynamically use it to create all the required zkeys files. In the circuit.sh file, `$1` is the variable name for accessing the `circuitName` parameter. This makes it easier for user to only supply the circuitName and the circom file to setup their own circuits.

File: examples/bri-3/zeroKnowledgeKeys/scripts/ptau.sh (hunk 1 above)
Comment (biscuitdey) [reply examples/bri-3/zeroKnowledgeKeys/scripts/ptau.sh:11]: We are using Groth16 for our circuit.

**Pull request merged**